django/core/management/templates.py

```diff
@@ -74,9 +76,14 @@ def handle(self, app_or_project, name, target=None, **options):
                 raise CommandError(e)
         else:
             top_dir = os.path.abspath(path.expanduser(target))
+            
             if not os.path.exists(top_dir):
-                raise CommandError("Destination directory '%s' does not "
-                                   "exist, please create it first." % top_dir)
+                # Creating recursive directories to store management commands is valid
+                if template_type == "command":
+                    os.makedirs(top_dir)
+                else: 
+                    raise CommandError("Destination directory '%s' does not "
+                                       "exist, please create it first." % top_dir)
```

Comment:
Shouldn't 'command' at least fail if the given app dir doesn't exist? Seems to me there should be a way of writing this code so 'command' isn't such a special case

Reply:
Ah! Yes, I can make the template dir include the `management/command` folder structure.